abdm_python_integrator/abha/views/abha_verification_views.py

```diff
@@ -1,102 +1,95 @@
-    resp = abdm_util.confirm_with_mobile_otp(otp, txn_id)
-    if "token" in resp:
-        resp = {"status": "success", "txnId": txn_id, "user_token": resp.get("token")}
-        resp.update(get_account_information(resp.get("user_token")))
-    return parse_response(resp)
-
-
-@api_view(["POST"])
-@permission_classes((IsAuthenticated,))
-@authentication_classes((app_settings.AUTHENTICATION_CLASS,))
-@required_request_params(["txn_id", "otp"])
-def confirm_with_aadhaar_otp(request):
-    txn_id = request.data.get("txn_id")
-    otp = request.data.get("otp")
-    resp = abdm_util.confirm_with_aadhaar_otp(otp, txn_id)
-    if "token" in resp:
-        resp = {"status": "success", "txnId": txn_id, "user_token": resp.get("token")}
-        resp.update(get_account_information(resp.get("user_token")))
-    return parse_response(resp)
-
-
-@api_view(["POST"])
-@permission_classes((IsAuthenticated,))
-@authentication_classes((app_settings.AUTHENTICATION_CLASS,))
-@required_request_params(["health_id"])
-def search_health_id(request):
-    health_id = request.data.get("health_id")
-    if app_settings.HRP_ABHA_REGISTERED_CHECK_CLASS is not None:
-        if (app_settings.HRP_ABHA_REGISTERED_CHECK_CLASS().
-                check_if_abha_registered(request.user, health_id)):
-            return generate_invalid_req_response(ABHA_ERROR_MESSAGES[ABHA_IN_USE_ERROR_CODE],
-                                                 error_code=ABHA_IN_USE_ERROR_CODE)
-    resp = abdm_util.search_by_health_id(health_id)
-    return parse_response(resp)
-
-
-@api_view(["POST"])
-@permission_classes((IsAuthenticated,))
-@authentication_classes((app_settings.AUTHENTICATION_CLASS,))
-@required_request_params(["user_token"])
-def get_health_card_png(request):
-    user_token = request.data.get("user_token")
-    return parse_response(abdm_util.get_health_card_png(user_token))
-
-
-@api_view(["POST"])
-@permission_classes((IsAuthenticated,))
-@authentication_classes((app_settings.AUTHENTICATION_CLASS,))
-@required_request_params(["health_id"])
-def get_existence_by_health_id(request):
-    health_id = request.data.get("health_id")
-    if app_settings.HRP_ABHA_REGISTERED_CHECK_CLASS is not None:
-        if app_settings.HRP_ABHA_REGISTERED_CHECK_CLASS().check_if_abha_registered(request.user, health_id):
-            return generate_invalid_req_response(ABHA_ERROR_MESSAGES[ABHA_IN_USE_ERROR_CODE],
-                                                 error_code=ABHA_IN_USE_ERROR_CODE)
-    resp = abdm_util.exists_by_health_id(health_id)
-    if "status" in resp:
-        resp = {"health_id": health_id, "exists": resp.get("status")}
-    return parse_response(resp)
+class GetAuthMethods(ABHABaseView):
+
+    def get(self, request, format=None):
+        health_id = request.query_params.get("health_id")
+        if not health_id:
+            error_msg = "Missing required parameter: health_id"
+            return generate_invalid_req_response(error_msg)
```

Comment:
Could this use `required_request_params`?

Reply:
required_request_params currently supports `POST` params, will modify it to support rhis. 

**FYI** - if time permits , I am considering to refactor request validations to use drf serializers, however that changes the error format.
That can ne addressed after implementing [this](https://github.com/dimagi/abdm-python-integrator/pull/1#discussion_r1326196603)

Reply:
Addressed in 8ec4dd581569bb0dfd333359b6bc78235f91d6e0